collection-scripts/gather_haproxy_config

```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+BASE_COLLECTION_PATH="must-gather"
+HAPROXY_CONFIG_PATH=${OUT:-"${BASE_COLLECTION_PATH}/haproxy"}
+
+mkdir -p "${HAPROXY_CONFIG_PATH}"/
+
+function gather_haproxy_config {
+    echo "INFO: Gathering HAProxy config files"
+    INGRESS_CONTROLLERS=$(oc get ingresscontroller -n openshift-ingress-operator --no-headers -o custom-columns=":metadata.name")
+    for IC in ${INGRESS_CONTROLLERS}; do
+        POD_STRING=$(oc get pods -n openshift-ingress --no-headers -o custom-columns=":metadata.name" --selector ingresscontroller.operator.openshift.io/deployment-ingresscontroller="${IC}")
+        PODS=($POD_STRING)
+        # Use first pod per ingress controller
+        oc cp openshift-ingress/"${PODS[0]}":/var/lib/haproxy/conf/haproxy.config "${HAPROXY_CONFIG_PATH}"/"${IC}"/haproxy.config
```

Comment:
You did `mkdir -p "${HAPROXY_CONFIG_PATH}"/` earlier, but you didn't `mkdir "${HAPROXY_CONFIG_PATH}"/"${IC}"/`...  But all right, it looks like `oc` takes care of making the directory: https://github.com/openshift/oc/blob/1bd1f5ff4aa98d6e4184c2948928b6111fa99191/vendor/k8s.io/kubectl/pkg/cmd/cp/cp.go#L466
So I think this is fine.

Reply:
Yea, I noticed the oc command did the directory creations for me, so I decided that I didn't need to explicitly make them.